### DISCUSSION TITLE
DB - Basic Search Engine Module full text search

### DISCUSSION THREAD

**Discussion by nomandera** [requarks/wiki#3380] [Help / Questions]
**Question**
Is full text search on any roadmap for the default install (e.g. docker) or will something like Elasticsearch, Solr or Sphinx always been required?

**Host Info:**
OS: [Docker]
Wiki.js version: [2.3.81]
Database engine: [Node.js 12.15.0 SQLite 3.30.1]

After some initial head scratching, assumptions on a bad setup and a reasonable about of searching I eventually realised that the basic Search Engine Module does not include full text search as documented here:

https://docs.requarks.io/en/search/db

Whilst I do not have vast experience in this area I find this quite unusual and limiting.

**Comment by winkelement** [comment]
A proper search engine should definitely be implemented through "external" modules in my opinion. Having a sophisticated search engine for a content heavy wiki would be a big task to implement in the software itself and i don't see the need to re-invent the wheel. 
I agree that for some users/administrators it is unexpected (despite being documented) that the "basic" search is so limited. 
To be honest i would suggest removing this "basic" search completely at the latest with version 2.5 when PostgreSQL becomes mandatory.

**Comment by alex-phillips** [comment]
I would argue that other wiki platforms support out-of-the-box search without needing an external search engine (MediaWiki, DokuWiki, Bookstack). I understand the performance and feature limitations that would come with a default 'basic' search, but implementing a search against the body of a wiki page the same way it searches against the title and description should be a basic feature included, at least in its simplest form.

**Comment by NGPixel** [comment]
My issue with enabling full text search on the basic search is that it's very DB intensive. If you include such functionality by default, most people won't bother switching to a proper search engine. In the end, they will have a slow and unreliable search experience (+ affecting the DB in general for all other operations).

Use PostgreSQL and you'll get a proper search engine without the need for an external one.

**Comment by nomandera** [comment]
I realise that there is an inherent scale and performance issue with using sqlite but I am not sure that removing functionality for those that fall below the point where scale or performance impacts usage is the correct approach.

I would suggest that the correct (and typical) approach would be to offer full functionality and mark it clearly for what it is; "_sqlite is useful only for micro and test installs_".

To offer a real world example, in my setup, a self hosted intranet wiki where the userbase consists of a small family install anything above sqlite would be needless extra complexity and patching burden.

**Comment by NGPixel** [comment]
Fair point. I'll look into adding it for 2.5 with a warning.

**Comment by alex-phillips** [comment]
@NGPixel looks like 2.5 is out but doesn't look like this was included. Any update on when we could expect this?

**Comment by KarelWintersky** [comment]
ManticoreSearch engine is a good (and simple) solution for full-text search.

But... how I can embed this to wiki.js engine?
